src/TfvcMigrator/Program.cs

```diff
@@ -117,16 +116,19 @@ private static RootPathChange ParseRootPathChange(string token)
         var unmappedAuthors = changesets.Select(c => c.Author)
             .Concat(changesets.Select(c => c.CheckedInBy))
             .Concat(allLabels.Select(l => l.Owner))
-            .Select(identity => identity.UniqueName)
-            .Where(name => !authorsLookup.ContainsKey(name))
-            .Distinct(StringComparer.OrdinalIgnoreCase)
+            .Where(identity => !authorsLookup.ContainsKey(identity.UniqueName))
+            .GroupBy(i => i.UniqueName)
+            .Select(grouping => grouping.First())
             .ToList();
 
         if (unmappedAuthors.Any())
         {
-            Console.WriteLine("An entry must be added to the authors file for each of the following TFVC users:");
+            Console.WriteLine("A valid email address must be added to the authors file for each of the following TFVC users:");
             foreach (var user in unmappedAuthors)
-                Console.WriteLine(user);
+            {
+                Console.WriteLine(user.UniqueName);
+                await File.AppendAllTextAsync(authors, $"{user.UniqueName} = {user.DisplayName} <email>{Environment.NewLine}");
```

Comment:
I expect File.AppendAllTextAsync should be sufficient, but only writing once with something like `string.Join(Environment.NewLine, unmappedAuthors.Select(author => $"{author.UniqueName} = {author.DisplayName} <email>"))`. Would that be okay? 

Do you have any good suggestions for detecting a trailing newline in a file? I'm having some trouble managing this because both StreamReader and the File class methods seem to bury trailing newlines.

Reply:
I'm okay with this. Part of me wants to lock the file for writing while reading and then append and then dispose the file stream, but I think this is a reflex that might be overcomplicating the scenario.

Reply:
I went ahead and implemented something like that. One thing I am on the fence about is manually disposing of the FileStream after the authors file has been written to. The way the code is as of the latest commit the file will stay locked until the MigrateAsync finishes - if I am not mistaken.